config/proposer/loader.go

```diff
@@ -0,0 +1,281 @@
+		if cliCtx.Bool(flags.EnableBuilderFlag.Name) {
+			psl.options.builderConfig = &validatorService.BuilderConfig{
+				Enabled:  true,
+				GasLimit: validator.Uint64(params.BeaconConfig().DefaultBuilderGasLimit),
+			}
+		}
+		return nil
+	}
+}
+
+// WithGasLimit applies the --suggested-gas-limit flag to proposer settings
+func WithGasLimit() SettingsLoaderOption {
+	return func(cliCtx *cli.Context, psl *SettingsLoader) error {
+		sgl := cliCtx.String(flags.BuilderGasLimitFlag.Name)
+		if sgl != "" {
+			gl, err := strconv.ParseUint(sgl, 10, 64)
+			if err != nil {
+				return errors.New("Gas Limit is not a uint64")
+			}
+			psl.options.gasLimit = reviewGasLimit(validator.Uint64(gl))
+			psl.options.hasGasLimit = true
+		}
+		return nil
+	}
+}
+
+// NewProposerSettingsLoader returns a new proposer settings loader that can process the proposer settings based on flag options
+func NewProposerSettingsLoader(cliCtx *cli.Context, db iface.ValidatorDB, opts ...SettingsLoaderOption) (*SettingsLoader, error) {
+	if cliCtx.IsSet(flags.ProposerSettingsFlag.Name) && cliCtx.IsSet(flags.ProposerSettingsURLFlag.Name) {
+		return nil, fmt.Errorf("cannot specify both %s and %s flags; choose one method for specifying proposer settings", flags.ProposerSettingsFlag.Name, flags.ProposerSettingsURLFlag.Name)
+	}
+	psExists, err := db.ProposerSettingsExists(cliCtx.Context)
+	if err != nil {
+		return nil, err
+	}
+	psl := &SettingsLoader{db: db, options: &flagOptions{}}
+
+	if cliCtx.IsSet(flags.SuggestedFeeRecipientFlag.Name) {
+		psl.loadMethods = append(psl.loadMethods, defaultFlag)
+	}
+	if cliCtx.IsSet(flags.ProposerSettingsFlag.Name) {
+		psl.loadMethods = append(psl.loadMethods, fileFlag)
+	}
+	if cliCtx.IsSet(flags.ProposerSettingsURLFlag.Name) {
+		psl.loadMethods = append(psl.loadMethods, urlFlag)
+	}
+	if len(psl.loadMethods) == 0 {
+		method := none
+		if psExists {
+			// override with db
+			method = onlyDB
+		}
+		psl.loadMethods = append(psl.loadMethods, method)
+	}
+
+	if psExists {
+		psl.existsInDB = true
+	}
+	for _, o := range opts {
+		if err := o(cliCtx, psl); err != nil {
+			return nil, err
+		}
+	}
+
+	return psl, nil
+}
+
+// Load saves the proposer settings to the database
+func (psl *SettingsLoader) Load(cliCtx *cli.Context) (*validatorService.ProposerSettings, error) {
+	var loadConfig *validatorpb.ProposerSettingsPayload
+
+	// override settings based on other options
+	if psl.options.builderConfig != nil && psl.options.hasGasLimit {
+		psl.options.builderConfig.GasLimit = psl.options.gasLimit
+	}
+
+	// check if database has settings already
+	if psl.existsInDB {
+		dbps, err := psl.db.ProposerSettings(cliCtx.Context)
+		if err != nil {
+			return nil, err
+		}
+		loadConfig = dbps.ToConsensus()
+	}
+
+	// start to process based on load method
+	for _, method := range psl.loadMethods {
+		switch method {
+		case defaultFlag:
+			suggestedFee := cliCtx.String(flags.SuggestedFeeRecipientFlag.Name)
```

Comment:
`suggestedFee` ==> `suggestedFeeRecipient` ?

(In our case, `suggestedFee` is a little bit like "we suggest a fee of xxx Gwei", which is (unfortunatly) not the case.)